lightning-invoice/src/utils.rs

```diff
@@ -519,14 +519,16 @@ where
 	fn send_payment(
 		&self, route: &Route, payment_hash: PaymentHash, payment_secret: &Option<PaymentSecret>
 	) -> Result<PaymentId, PaymentSendFailure> {
-		self.send_payment(route, payment_hash, payment_secret)
+		let payment_id = PaymentId(payment_hash.0);
```

Comment:
Might be a dumb question, but why do we always use the hash here instead of adding a payment ID param to the above signature in the trait?

When we deal with the Payer API is it that we don't want to expose these internal things or just not to break stuff downstream?